packages/web-app-files/tests/unit/views/SharedViaLink.spec.js

```diff
@@ -1,29 +1,194 @@
+        Promise.resolve({
+          ocs: {
+            data: activeFiles
+          }
+        })
+      )
+    })
+  )
+}
+
+const stubs = {
+  'router-link': true,
+  'no-content-message': true,
+  pagination: true,
+  'oc-table-files': true,
+  'context-actions': true,
+  'list-info': true
+}
+
+const selectors = {
+  listLoader: 'list-loader-stub',
+  noContentMessage: '#files-shared-via-link-empty',
+  ocTableFiles: '#files-shared-via-link-table',
+  contextActions: 'context-actions-stub',
+  listInfo: 'list-info-stub'
+}
+
 describe('SharedViaLink view', () => {
+  jest
+    .spyOn(MixinFilesListPagination.methods, '$_filesListPagination_updateCurrentPage')
+    .mockImplementation()
+  const spyAdjustTableHeaderPosition = jest.spyOn(
+    MixinFilesListPositioning.methods,
+    'adjustTableHeaderPosition'
+  )
+  const spyTriggerDefaultAction = jest
+    .spyOn(FileActions.methods, '$_fileActions_triggerDefaultAction')
+    .mockImplementation()
+  const spyLoadResources = jest.spyOn(SharedViaLink.methods, 'loadResources')
+  const spyRowMounted = jest.spyOn(SharedViaLink.methods, 'rowMounted')
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
   describe('when the page has loaded successfully', () => {
-    it.todo('should load the resources')
-    it.todo('should adjust the table header position')
+    beforeEach(async () => {
+      getShallowWrapper()
+      await flushPromises()
+    })
+
+    it('should load the resources', () => {
+      expect(spyLoadResources).toHaveBeenCalledTimes(1)
+    })
+    it('should adjust the table header position', () => {
+      expect(spyAdjustTableHeaderPosition).toHaveBeenCalledTimes(1)
+    })
   })
 
   describe('when the view is still loading', () => {
-    it.todo('should show list-loader component')
+    it('should show list-loader component', () => {
+      const wrapper = getShallowWrapper()
+      const listLoader = wrapper.find(selectors.listLoader)
+
+      expect(listLoader.exists()).toBeTruthy()
+      expect(listLoader).toMatchSnapshot()
+    })
   })
 
   describe('when the view is not loading anymore', () => {
-    it.todo('should not show list-loader component')
+    it('should not show list-loader component', async () => {
+      const wrapper = getShallowWrapper()
+      await flushPromises()
+
+      expect(wrapper.find(selectors.listLoader).exists()).toBeFalsy()
+    })
 
     describe('when there are no files to be displayed', () => {
-      it.todo('should show no-content-message component')
-      it.todo('should not show oc-table-files component')
+      let wrapper
+      beforeEach(async () => {
+        stubs['no-content-message'] = false
+        wrapper = getWrapper()
+        await flushPromises()
+      })
+
+      it('should show no-content-message component', () => {
+        const noContentMessage = wrapper.find(selectors.noContentMessage)
+
+        expect(noContentMessage.exists()).toBeTruthy()
+        expect(noContentMessage).toMatchSnapshot()
```

Comment:
It feels a bit odd testing snapshot of a child component in the parent instead of tests of that component...